Mu2eG4/src/Mu2eG4MTRunManager.cc

```diff
@@ -118,71 +121,49 @@ namespace mu2e {
     }
 
     /////////////////////////////////////////////////////////////////////////////////////
-    /////////////////////////////////////////////////////////////////////////////////////
-    //RANDOM NUMBER SEEDING: NEED TO FIGURE THIS OUT
-    //taken from G4MTRunManager::InitializeEventLoop()
-    int numevents = maxNumEventstoSeed_;
+    //RANDOM NUMBER SEEDING
+    SetEventModulo(1);//this sets eventModuloDef
+    int seedbunchsize = 10000;
     int numworkers = 1;
-    SetNumberOfEventsToBeProcessed(numevents);
-    //numberOfEventToBeProcessed = 1000;//this needs to be filled from the # of events we are processing
+    numberOfEventToBeProcessed = std::numeric_limits<int>::max();
     numberOfEventProcessed = 0;
 
     nSeedsUsed = 0;
     nSeedsFilled = 0;
 
     if(verboseLevel>0)
       { timer->Start(); }
-
-    //initialize seeds
-    //If user did not implement InitializeSeeds,
-    // use default: nSeedsPerEvent seeds per event
-    //eventModuloDef = 0 by default
-    if( eventModuloDef > 0 ) {
-      eventModulo = eventModuloDef;
-      if(eventModulo > numberOfEventToBeProcessed/numworkers) {
-        eventModulo = numberOfEventToBeProcessed/numworkers;
-        if(eventModulo<1) eventModulo =1;
-
-        G4ExceptionDescription msgd;
-        msgd << "Event modulo is reduced to " << eventModulo
-             << " to distribute events to all threads.";
-        G4Exception("G4MTRunManager::InitializeEventLoop()", "Run10035", JustWarning, msgd);
-      }
-    }
-    else {
-      eventModulo = int(std::sqrt(double(numberOfEventToBeProcessed/numworkers)));
-      if(eventModulo<1) eventModulo = 1;
-    }
-
-    if ( InitializeSeeds(numevents) == false && numevents>0 ) {
+    
+    //if we are using G4's seed filling scheme
+    if ( InitializeSeeds(seedbunchsize) == false && seedbunchsize>0 ) {
 
       G4RNGHelper* helper = G4RNGHelper::GetInstance();
```

Comment:
Is this thread safe?

Reply:
No, but this code is in the call initializeG4 which is called during beginRun when art is still running single-threaded.